File: Python/PythonKatas/OddEven/oddeven.py
import math

class OddEven(object):

    Even = ("""Even""")
    Odd = ("""Odd""")

    def convert(number):
        if number % 2 == 0:
            return OddEven.Even
        if OddEven.__is_prime(number):
            return number

        return OddEven.Odd

    def convert_range(start, stop):
        return ' '.join(str(OddEven.convert(x)) for x in range(start, stop))

    def convert_with_string(number):
        if (str(number).endswith("0") or 
            str(number).endswith("2") or 
            str(number).endswith("4") or 
            str(number).endswith("6") or 
            str(number).endswith("8")):
            return OddEven.Even
        
        if OddEven.__is_prime(number):
            return number

        return OddEven.Odd

    def __is_prime(n):
        if n == 2 or n == 3: return True
        if n%2 == 0 or n < 2: return False
        for i in range(3,int(n**0.5)+1,2):
            if n%i == 0:
                return False    

        return True

File: TypeScript/TypeScriptKatas/OddEven/oddeven.ts
class OddEven {
    
    even = "Even";
    odd = "Odd";

    convert(num: number): string {

        if (num % 2 === 0) return this.even;
        if (this.isPrime(num)) return num.toString();

        return this.odd;
    }

    convertRange(start: number, end: number): string {

        var result = "";

        for (let i = start; i <= end; i++) {
            result += this.convert(i) + " ";
        }

        return result.trim();
    }

    convertWithString(num: number): string {

        var numString = num.toString();

        if (StringUtilities.endsWith(numString, "0") ||
            StringUtilities.endsWith(numString, "2") ||
            StringUtilities.endsWith(numString, "4") ||
            StringUtilities.endsWith(numString, "6") ||
            StringUtilities.endsWith(numString, "8"))
            return this.even;

        return this.isPrime(num) ? numString : this.odd;
    }

    private isPrime(num: number): boolean {

        if (num === 1) return false;

        for (let i = 3; i <= Math.sqrt(num); i += 2) {
            if (num % i === 0) return false;
        }

        return true;
    }
}

class StringUtilities {

    static endsWith(source: string, value: string): boolean {

        return source.substring(source.length - value.length, source.length) === value;
    }
}